eclair-core/src/main/scala/fr/acinq/eclair/api/NewService.scala

```diff
@@ -0,0 +1,251 @@
+package fr.acinq.eclair.api
+
+import akka.http.scaladsl.server._
+import fr.acinq.bitcoin.Crypto.PublicKey
+import fr.acinq.bitcoin.{ByteVector32, MilliSatoshi, Satoshi}
+import fr.acinq.eclair.{Kit, ShortChannelId}
+import FormParamExtractors._
+import akka.NotUsed
+import akka.actor.{Actor, ActorRef, ActorSystem, Props}
+import akka.http.scaladsl.model.HttpMethods.POST
+import akka.http.scaladsl.model.{ContentTypes, HttpRequest, HttpResponse, StatusCodes}
+import akka.http.scaladsl.model.headers.CacheDirectives.{`max-age`, `no-store`, public}
+import akka.http.scaladsl.model.headers.{`Access-Control-Allow-Headers`, `Access-Control-Allow-Methods`, `Cache-Control`}
+import akka.http.scaladsl.model.ws.{Message, TextMessage}
+import akka.http.scaladsl.server.directives.{Credentials, LoggingMagnet}
+import akka.stream.{ActorMaterializer, OverflowStrategy}
+import akka.stream.scaladsl.{BroadcastHub, Flow, Keep, Source}
+import fr.acinq.eclair.payment.{PaymentLifecycle, PaymentReceived, PaymentRequest}
+import grizzled.slf4j.Logging
+import scodec.bits.ByteVector
+import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.duration._
+
+trait NewService extends Directives with Logging with MetaService {
+
+  // important! Must NOT import the unmarshaller as it is too generic...see https://github.com/akka/akka-http/issues/541
+  import JsonSupport.marshaller
+  import JsonSupport.formats
+  import JsonSupport.serialization
+
+  def password: String
+
+  def eclairApi: EclairApi
+
+  implicit val actorSystem: ActorSystem
+  implicit lazy val ec = actorSystem.dispatcher
+  implicit val mat: ActorMaterializer
+
+  // a named and typed URL parameter used across several routes, 32-bytes hex-encoded
+  val channelIdNamedParameter = "channelId".as[ByteVector32](sha256HashUnmarshaller)
```

Comment:
Centralizing `channelId` parameter in a variable is good for maintenance, but when reading the file I cannot easily get the endpoints' parameters in a quick glance, and I have to look up what `channelIdNamedParameter` means.

We should either:
- have a consistent and exhaustive way of describing the endpoint parameter:
```scala
formFields("channelId".as[ByteVector32](sha256HashUnmarshaller), "scriptPubKey".as[ByteVector](binaryDataUnmarshaller).?)...
```

- or at least rename `channelIdNamedParameter` to `channelId` so that we know that the `close` endpoint requires the `channelId` param

Reply:
Fixed in e07e9f6